### DISCUSSION TITLE
Community/Members papers section

### DISCUSSION THREAD

**Discussion by FCLC** [hpc-social/hpc-social.github.io#46] [Ideas]
Based on https://github.com/hpc-social/blog/issues/3, idea is having a sort of mini arxiv for the HPC community that serves to hold papers, posters, talks etc. in a common directory. 

Ideas for organizing in a clean, accessible way is to make use of the existing work behind the ar5iv project that converts latex posts into web compatible HTML5 posts. 

example and information on Ar5iv:

https://ar5iv.labs.arxiv.org/html/1910.06709
https://ar5iv.labs.arxiv.org/
https://blog.arxiv.org/2022/02/21/arxiv-articles-as-responsive-web-pages/

Converts Latex to what is basically an HTML5 compliant webformat.

For hosting: posters, papers and articles we're fine (the format has the bonus that it works super well for things like screen readers on the accessibility front 🙌🏻)

For talks I don't see a world where we're hosting our own full video CDN unless the funding is significant. Youtube or the likes is more than fit for purpose and probably better in terms of bringing in new people to the community, especially students.

**Comment by vsoch** [comment]
Pinging @davidbeckingsale - we actually just were talking about this today, and maybe it could be a collaborative effort! I think we'd want to take a shot at not just giving a home for "non sexy research software that other journals to be unnamed would not call research software" but also a way to rethink credit and value. I had a server-based idea around citelang, which asserts that the source of truth for software should be in package managers, of course I haven't built anything yet. Of course I could! <3  

So I definitely want to think more about this idea and (maybe?) if we can come to a cool idea can make it over the holiday break.

**Reply by vsoch** [reply]
I guess my thing is that mastodon (at least the UI on my computer / phone) is just kind of slow? it's not pleasant to use.

**Reply by FCLC** [reply]
wow, sorry, phone posted in the wrong thread :sweat_smile:

**Reply by vsoch** [reply]
haha no worries, I got it!

**Comment by alansill** [comment]
There are several people (underfox, Tadashi Ogawa, others) who routinely post reviews or summaries of articles, papers, even patents) that they find interesting. These often have interesting commentary on the points that make the item(s) interesting. Given that things like Academia.edu, Research Gate, etc. exist where users can post their publications for others to see and follow, how would we differentiate such a service and add useful features (for example, commentary or question/answer sections, etc.?)

**Reply by vsoch** [reply]
My vision slash dream would be to have something that looks like arxiv in terms of professionalism and LaTex, but provide (what I call) "living documents." Let me explain!

This idea would be a (on a high level) "place for papers" and use this idea that has always fascinated me - the idea of a living publication. E.g., maybe you write a PDF, you put it online, but you change it as the state of the world changes. In this light I imagine something like an ArXiV, but optimized for people giving feedback, and (akin to arxiv) being able to update it when necessary. In terms of the UI itself, this would mean we optimize for people to interact with / review the text (there are platforms people have made to do this, but they never really took off for the use cases presented). Then the site would provide the identifier to link to for a citation, serving the purpose of a journal. You'd add it to your CITATION.cff in your GitHub repo just like a journal, but unlike traditional journals that have huge costs and "behind the barrier of reviewer 3" the work would be totally open and public, and the community that uses the server (or anyone on the internet really) would be able to be critical of it, etc. It would be a constant, ongoing review, with opportunity for someone to reproduce (fork) a paper, or open an issue and have it worked on and discussed, or change. Actually now that I think of it, the integration would work really nicely between GitHub (version control, issues, PRs) and ArXiv (pretty display of the current state).

So we can scope that to "something in LaTex that needs a home and a DOI and some cool stuff around it to make peer review more open and rolling" And the decision to cite something is up to the person that reads the paper. If they think it's good enough, sound, they cite it. Otherwise they don't have to, they can give critical feedback. There is no artificial requirment of three random humans to say "yes this is good enough" that is for the reader to decide. It would make people citing more responsible for understanding what they are citing, and thus probably asking questions or making comments if something feels off.

This platform (for  me) would also fill a niche that isn't addressed - the "non mathy and non domain specific unsexy side of research software" that is the space of container technologies, databases, APIs, orchestration, etc. that has been explicitly designed for research, but many academics that run journals would turn their nose up and say "This is not research software reject" and keep up their gatekeeping nonsense. We need a space where these projects can thrive, because (ironically) they tend to be _more_important for the research ecosystem than say, the 5 millionth set of python scripts (that do fit in the box and are accepted) to do genomic analysis the same way, again.

**Reply by alansill** [reply]
In principle this is exactly the need that OGF has developed a process to address for nearly two decades that led to creation of the grid. At its height, there were hundreds of contributors working in dozens of working groups, research groups, and community groups with regular meetings all the time and a process that not only encouraged but required community input and review as well as internal steps designed not to hold back but to ensure that such community review took place. Existing documents could be revised if changes amounted only to errata or non-breaking changes, or formally replaced with new versions with a new document number and traceability if the changes introduced were greater than that level, to preserve citeability for existing implementations. OGF focused on community consensus documents, leaving implementation to the community, but requiring at least two independent implementations to promote a proposed recommendation to a full one (the highest category of its publications) with documentation of extended adoption. There are informational, experimental, and community practice documents in the system as well. 

I'm not saying we should directly use OGF but can build something like its model. If you are interested, take a look at https://docs.ogf.org and note that we've recently introduced GitBook as an additional option for document development and publication. 

I plan to use this process to produce the open community document for the recent DAAC 2022 workshop on "Reducing HPC's Carbon Footprint".

**Reply by vsoch** [reply]
Interesting, although arguably if I've never heard of it (let alone use it) it's not meeting the need that I have.

**Reply by alansill** [reply]
Um, ouch.

**Reply by vsoch** [reply]
I'm sorry I didn't mean that to be ouch, I just haven't heard of this until now!

**Reply by alansill** [reply]
NP, just an organization I've been volunteering with for the past 20 years or so!  ;-). It's one of the sponsors for hpc.social btw. (We're in the process of modernizing our infrastructure and a partnership with hpc.social also made sense.)

In this case I wasn't so much pointing out the specific infrastructure, which was pretty much specific to grids, as the problems of hosting "live" documents when you want them to be the basis for work in he community and how that organization managed things to allow documents to be proposed, commented on, reviewed, published, and maintianed with options for revision and/or replacement. The full process is described in [https://ogf.org/documents/GFD.152.pdf](https://ogf.org/documents/GFD.152.pdf). It's also a good example of an open participant-driven organization.

Returning to your original concept, a totally open live document could just be a web site, I suppose. If you want process around that for how to propose, vet, review, and publish updates that could mean new tooling but the principles woud be good to establish ahead of time. Since the formation of grids, the process of publishing open specifications for people to work from has been expanded greatly by other organizations including just putting things out on repositories as individuals to follow (this comment is specific to specifications but applies in general to any document for which you want an organized process for revision). 

I like the idea of live documents but you need a process for revision if the changes to the document will produce changes in how people refer to or use its contents. At some level, any document-production software will face the same issues. In summary, it depends on what you mean by "publication" and whether a work product that is subject to revision can have a place in the chain of references for a topic or for how to do something if it is subject to change at any time, and if so, how to manage that revision process.

**Reply by vsoch** [reply]
Makes total sense! I think what I’m imagining is more scoped to live publication - the “living documentation” is an idea I came up with in grad school that generally says something is continually being updated (but in this context not explicitly docs).
